### PR TITLE
Fix saving and loading pickle files on Big Endian systems

### PR DESCRIPTION
**Pull request by AlekseiNikiforovIBM**
This change fixes test/test_cpp_api_parity.py tests on Big Endian systems.

### CODE REVIEW

File: torch/csrc/jit/serialization/unpickler.cpp [hunk 1]
```diff
@@ -316,21 +317,21 @@ PickleOpCode Unpickler::readInstruction() {
       stack_.emplace_back(int64_t(value));
     } break;
     case PickleOpCode::BININT2: {
-      uint16_t value = read<uint16_t>();
+      uint16_t value = le16toh(read<uint16_t>());
       stack_.emplace_back(int64_t(value));
     } break;
     case PickleOpCode::BININT: {
-      int32_t value = read<int32_t>();
+      int32_t value = le32toh(read<int32_t>());
```
Comment (malfet) [inline torch/csrc/jit/serialization/unpickler.cpp:324]: Hmm, wouldn't it raise the warning, as `le32toh` expects unsigned input and returns unsigned type as result

**Review by malfet — APPROVED** [review]

File: torch/csrc/jit/serialization/unpickler.cpp (hunk 1 above)
Comment (AlekseiNikiforovIBM) [reply torch/csrc/jit/serialization/unpickler.cpp]: No warnings here for me. It's converted implicitly.

**Comment by AlekseiNikiforovIBM** [comment]
Had to rework it a bit because `endian.h` is present only on Linux. Please let me know if it'd be better to rework it some other way.

There are already `THP_encode/decodeInt*Buffer`  functions present, but using them makes code look really bloated.

**Comment by kiszk** [comment]
I think that #93146 and this PR addresses the same issue.

**Comment by AlekseiNikiforovIBM** [comment]
Yes, looks so.

**Comment by kit1980** [comment]
Is this ready to merge?

**Comment by kiszk** [comment]
You may want to rebase this branch with the latest master, and then push this branch again. so, tests will be re-started.

**Pull request closed without merging**